jira/src/test/java/org/arquillian/extension/governor/jira/impl/reporter/JiraGovernorRecorderTest.java

```diff
@@ -0,0 +1,160 @@
+package org.arquillian.extension.governor.jira.impl.reporter;
+
+import org.arquillian.extension.governor.api.detector.Detector;
+import org.arquillian.extension.governor.api.detector.utils.OS;
+import org.arquillian.extension.governor.jira.api.Jira;
+import org.arquillian.extension.governor.jira.configuration.JiraGovernorConfiguration;
+import org.arquillian.recorder.reporter.PropertyEntry;
+import org.arquillian.recorder.reporter.event.PropertyReportEvent;
+import org.arquillian.recorder.reporter.model.entry.KeyValueEntry;
+import org.arquillian.recorder.reporter.model.entry.table.TableCellEntry;
+import org.arquillian.recorder.reporter.model.entry.table.TableEntry;
+import org.arquillian.recorder.reporter.model.entry.table.TableRowEntry;
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.test.spi.event.suite.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JiraGovernorRecorderTest {
```

Comment:
I think these tests need some clean up. They assert too much and have quite mixed flow which is hard to follow. Can you try to distill them to some concrete features we would like to verify and have simple flow of

  * setup initial context for the test
  * exercise components under tests
  * verify state or behaviour using assertions or mock verifications at the end?

I would also recommend to use [assertj](http://joel-costigliola.github.io/assertj/) for fluent and readable assertions.